src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/WorkloadManifestReader.SystemTextJson.cs

```diff
@@ -0,0 +1,113 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#if USE_SYSTEM_TEXT_JSON
+
+using System;
+using System.Buffers;
+using System.IO;
+using System.Text.Json;
+
+namespace Microsoft.NET.Sdk.WorkloadManifestReader
+{
+    internal partial class WorkloadManifestReader
+    {
+        public static WorkloadManifest ReadWorkloadManifest(Stream manifestStream)
+        {
+            var readerOptions = new JsonReaderOptions
+            {
+                AllowTrailingCommas = true,
+                CommentHandling = JsonCommentHandling.Skip
+            };
+
+            var reader = new Utf8JsonStreamReader(manifestStream, readerOptions);
+
+            return ReadWorkloadManifest(ref reader);
+        }
+
+        private ref struct Utf8JsonStreamReader
+        {
+            static ReadOnlySpan<byte> utf8Bom => new byte[] { 0xEF, 0xBB, 0xBF };
+
+            const int segmentSize = 4096;
+
+            Utf8JsonReader reader;
+            readonly Stream stream;
+
+            IMemoryOwner<byte> buffer;
+
+            Span<byte> span;
+
+            public Utf8JsonStreamReader(Stream stream, JsonReaderOptions readerOptions)
+            {
+                this.stream = stream;
+
+                buffer = MemoryPool<byte>.Shared.Rent(segmentSize);
+                var readCount = stream.Read(buffer.Memory.Span);
+                span = buffer.Memory.Slice(0, readCount).Span;
+
+                if (span.StartsWith(utf8Bom))
+                {
+                    span = span.Slice(utf8Bom.Length, span.Length - utf8Bom.Length);
+                }
```

Comment:
Yes, System.Text.Json does not consider the BOM to be a valid part of UTF8 input.

> Since the Utf8JsonReader considers the input to be JSON text, a UTF-8 byte order mark (BOM) is considered invalid JSON. The caller needs to filter that out before passing the data to the reader.

https://docs.microsoft.com/en-us/dotnet/standard/serialization/system-text-json-migrate-from-newtonsoft-how-to#read-with-a-stream-or-pipereader